edward/util/random_variables.py

```diff
@@ -778,3 +778,66 @@ def transform(x, *args, **kwargs):
   new_x = TransformedDistribution(x, bij, *args, **kwargs)
   new_x.support = new_support
   return new_x
+
+
+def marginal(x, n):
+  """Performs a full graph sample on the provided random variable.
+
+  Given a random variable and a sample size, adds an additional sample
+  dimension to the root random variables in x's graph, and samples from
+  a new graph in terms of that sample size.
+
+  Args:
+    x : RandomVariable.
+      Random variable to perform full graph sample on.
+    n : tf.Tensor or int
+      The size of the full graph sample to take.
+
+  Returns:
+    tf.Tensor.
+    The fully sampled values from x, of shape [n] + x.shape
```

Comment:
`x.shape = x.sample_shape + x.batch_shape + x.event_shape`. You replace the sample_shape, so the output should have shape `[n] + x.batch_shape + x.event_shape`. But I guess it currently fails if `x.sample_shape` is non-scalar anyways.